src/core/StepRecorder.cpp

```diff
@@ -288,27 +288,28 @@ void StepRecorder::removeNotesReleasedForTooLong()
 	int nextTimout = std::numeric_limits<int>::max();
 	bool notesRemoved = false;
 
-	QMutableVectorIterator<StepNote*> itr(m_curStepNotes);
-	while (itr.hasNext())
+	for (const auto& stepNote : m_curStepNotes) 
 	{
-		StepNote* stepNote = itr.next();
-
-		if(stepNote->isReleased())
+		if (stepNote->isReleased()) 
 		{
-			const int timeSinceReleased = stepNote->timeSinceReleased(); // capture value to avoid wraparound when calculting nextTimout
-			if (timeSinceReleased >= REMOVE_RELEASED_NOTE_TIME_THRESHOLD_MS)
+			const int timeSinceReleased = stepNote->timeSinceReleased();
+			if (timeSinceReleased >= REMOVE_RELEASED_NOTE_TIME_THRESHOLD_MS) 
 			{
 				delete stepNote;
-				itr.remove();
 				notesRemoved = true;
 			}
-			else
+			else 
 			{
 				nextTimout = min(nextTimout, REMOVE_RELEASED_NOTE_TIME_THRESHOLD_MS - timeSinceReleased);
 			}
 		}
 	}
 
+	m_curStepNotes.erase(std::remove_if(m_curStepNotes.begin(), m_curStepNotes.end(), [](auto stepNote) 
+	{ 
+		return stepNote->timeSinceReleased() >= REMOVE_RELEASED_NOTE_TIME_THRESHOLD_MS; 
```

Comment:
Aren't we deleted stuff in the loop above? This may result in use after free.

Reply:
Honestly don't know how I didn't see that :sweat_smile:. I'll write it differently.